### PR TITLE
Questions: changes in layout and workflow, bug fixing

### PR DESCRIPTION
**Pull request by differentreality**
https://github.com/openSUSE/osem/issues/551
Includes https://github.com/openSUSE/osem/pull/648
Fixes https://github.com/openSUSE/osem/issues/613

Small change in layout of _form.
Small change in workflow: Question can be added without answers. But a question can be associated with a conference only after answers are added.

### CODE REVIEW

File: app/controllers/admin/questions_controller.rb [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
     load_and_authorize_resource :conference, find_by: :short_title
-    load_and_authorize_resource through: :conference, except: [:new, :create]
+    load_and_authorize_resource except: [:create]
 
     def index
       authorize! :index, Question.new(conference_id: @conference.id)
-      @questions = Question.where(global: true).all | Question.where(conference_id: @conference.id)
-      @questions_conference = @conference.questions
-      @new_question = @conference.questions.new
+      @questions = Question.where(global: true).all | Question.where(conference_id: @conference.id) | @conference.questions
```
Comment (ChrisBr) [inline app/controllers/admin/questions_controller.rb]: What's the difference between 
``Question.where(conference_id: @conference.id) ``
and 
``@conference.questions``

File: app/controllers/admin/questions_controller.rb [hunk 2]
```diff
@@ -22,45 +21,76 @@ def new
...
+    def edit; end
 
     # PUT questions/1
     def update
-      if @question.update_attributes(params[:question])
-        redirect_to(admin_conference_questions_path(conference_id: @conference.short_title), notice: "Question '#{@question.title}' for #{@conference.short_title} successfully updated.")
+      @question.assign_attributes(params[:question])
+
+      if @question.question_type == QuestionType.find_by(title: 'Yes/No')
+        @question.answers = [ Answer.find_or_create_by(title: 'Yes'), Answer.find_or_create_by(title: 'No') ]
```
Comment (ChrisBr) [inline app/controllers/admin/questions_controller.rb]: DRY, see line 31! 
Can't we do this in a callback?

File: app/controllers/admin/questions_controller.rb [hunk 3]
```diff
@@ -22,45 +21,76 @@ def new
...
+      end
+
+      if @question.save
+        if @question.answers.blank?
```
Comment (ChrisBr) [inline app/controllers/admin/questions_controller.rb]: Huh? You save it and then delete it? And without telling the user what happened?

**Comment by ChrisBr** [comment]
You also need to rebase! Otherwise good work!

**Comment by ChrisBr** [comment]
What's the actual state of this one?

**Comment by differentreality** [comment]
Henne had suggested a full refactoring (https://github.com/openSUSE/osem/issues/551) which I don't know how to implement after a certain point (hence my last comment in the issue). So I am not sure which way we want to proceed. We also very briefly discussed the option of using a gem, but we haven't quite decided on one, or had any further communication about the whole issue.

**Comment by hennevogel** [comment]
The whole Question/Answer thing is a mess I would like to get rid of by using some gem to build questionnaires. I'm not sure of the implications for this PR...

**Comment by Rolence** [comment]
You can't just get rid of it and use gems to generate questions. This will not be in flow with the grand design  of the whole system

**Comment by ChrisBr** [comment]
@differentreality @hennevogel what's the consent of this PR and the Question/Answer thing?

**Comment by ChrisBr** [comment]
If there is no consent yet, can we just close the PR for now?

**Comment by hennevogel** [comment]
We can yes

File: app/controllers/admin/questions_controller.rb (hunk 1 above)
Comment (differentreality) [inline app/controllers/admin/questions_controller.rb]: https://github.com/openSUSE/osem/blob/master/app/models/question.rb#L3
From the Question model:
`has_and_belongs_to_many :conferences`
This relationship is used for enabled questions (which will show up in registration)

The conference_id attribute is a manual one, used to know which questions were created for that conference, even if they are not enabled.

**Pull request closed without merging**